openmls/src/tree/secret_tree.rs

```diff
@@ -182,16 +191,32 @@ impl SecretTree {
 
         let handshake_ratchet_secret =
             node_secret.kdf_expand_label(backend, "handshake", b"", ciphersuite.hash_length())?;
-        let handshake_sender_ratchet = SenderRatchet::new(index, &handshake_ratchet_secret);
-        self.handshake_sender_ratchets[index.as_usize()] = Some(handshake_sender_ratchet);
         let application_ratchet_secret = derive_tree_secret(
             node_secret,
             "application",
             0,
             ciphersuite.hash_length(),
             backend,
         )?;
-        let application_sender_ratchet = SenderRatchet::new(index, &application_ratchet_secret);
+        let (handshake_sender_ratchet, application_sender_ratchet) = if index == self.own_index {
```

Comment:
This is the only place where `own_index` is used. It feels like a lot of overhead carrying the value around for this one check.

Reply:
👍🏻  yeah that's nice. We could pass in something to distinguish the own index instead. But let's keep it as is.